File: fixtures/ts-basic/src/repositories/userRepository.ts
import { User, CreateUserInput } from "../models/user";

export class UserRepository {
  private users = new Map<string, User>();

  findById(id: string): User | undefined {
    return this.users.get(id);
  }

  findByEmail(email: string): User | undefined {
    for (const user of this.users.values()) {
      if (user.email === email) return user;
    }
    return undefined;
  }

  insert(input: CreateUserInput, passwordHash: string): User {
    const user: User = {
      id: crypto.randomUUID(),
      email: input.email,
      name: input.name,
      passwordHash,
      role: "member",
      createdAt: new Date(),
    };
    this.users.set(user.id, user);
    return user;
  }

  all(): User[] {
    return Array.from(this.users.values());
  }
}

File: fixtures/csharp-basic/Repositories/UserRepository.cs
using System.Collections.Generic;
using OrderSystem.Models;

namespace OrderSystem.Repositories;

public class UserRepository : IUserRepository
{
    private readonly List<User> _users = new List<User>();

    public User FindByEmail(string email)
    {
        foreach (var user in _users)
        {
            if (user.Email == email)
            {
                return user;
            }
        }
        return null;
    }

    public User FindById(int id)
    {
        foreach (var user in _users)
        {
            if (user.Id == id)
            {
                return user;
            }
        }
        return null;
    }

    public void Save(User user)
    {
        _users.Add(user);
    }
}
